src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java

```diff
@@ -331,6 +335,23 @@ public Commit getLastCommitDetails(Object requestBody) throws VersionControlExce
         return commit;
     }
 
+    @Override
+    public ZonedDateTime getPushDate(ProgrammingExerciseParticipation participation, String commitHash) {
+        try {
+            String repositoryUrl = urlService.getRepositoryPathFromRepositoryUrl(participation.getVcsRepositoryUrl());
+            List<Event> events = gitlab.getEventsApi().getProjectEvents(repositoryUrl, Constants.ActionType.PUSHED, null, null, null, Constants.SortOrder.DESC);
```

Comment:
If I understand it correctly, no limit restriction is implemented for GitLab, right?

Reply:
Please recheck